ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java

```diff
@@ -86,75 +88,80 @@ public Object parse(Object o) {
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
-        @SuppressWarnings("unchecked")
-        List<String> indexList = parameters.containsKey("index")
-            ? gson.fromJson(parameters.get("index"), List.class)
-            : Collections.emptyList();
-        if (indexList.isEmpty()) {
-            @SuppressWarnings("unchecked")
-            T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
-            listener.onResponse(empty);
-            return;
-        }
+        try {
+            List<String> indexList = new ArrayList<>();
+            if (StringUtils.isNotBlank(parameters.get("index"))) {
+                indexList = parameters.containsKey("index") ? gson.fromJson(parameters.get("index"), List.class) : Collections.emptyList();
+            }
 
-        final String[] indices = indexList.toArray(Strings.EMPTY_ARRAY);
+            if (indexList.isEmpty()) {
+                @SuppressWarnings("unchecked")
+                T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
+                listener.onResponse(empty);
+                return;
+            }
 
-        final IndicesOptions indicesOptions = IndicesOptions.strictExpand();
-        final boolean local = Boolean.parseBoolean(parameters.get("local"));
-        final TimeValue clusterManagerNodeTimeout = DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT;
+            final String[] indices = indexList.toArray(Strings.EMPTY_ARRAY);
 
-        ActionListener<GetIndexResponse> internalListener = new ActionListener<GetIndexResponse>() {
+            final IndicesOptions indicesOptions = IndicesOptions.strictExpand();
+            final boolean local = Boolean.parseBoolean(parameters.get("local"));
```

Comment:
If "local" is not present, it silently defaults to false, which may not be what you want. Better version if you want to handle defaults explicitly
final boolean local = Boolean.parseBoolean(parameters.getOrDefault("local", "false"));